span.go

```diff
@@ -58,3 +62,30 @@ type Span interface {
 	// TraceContext.
 	AddToGoContext(goCtx context.Context) (Span, context.Context)
 }
+
+// See Span.Log(). Every LogData instance should specify at least one of Event
+// and/or Payload.
+type LogData struct {
+	// The timestamp of the log record; if set to the default value (the unix
+	// epoch), implementations should use time.Now() implicitly.
+	Timestamp time.Time
+
+	// Event (if non-empty) should be the stable name of some notable moment in
+	// the lifetime of a Span. For instance, a Span representing a browser page
+	// load might add an Event for each of the Performance.timing moments
+	// here: https://developer.mozilla.org/en-US/docs/Web/API/PerformanceTiming
+	//
+	// While it is not a formal requirement, Event strings will be most useful
+	// if they are *not* unique; rather, tracing systems should be able to use
+	// them to understand how two similar Spans relate from an internal timing
+	// perspective.
+	Event string
+
+	// Payload is a free-form potentially structured object which Tracer
+	// implementations may retain and record all, none, or part of.
+	//
+	// If included, `Payload` should be restricted to data derived from the
+	// instrumented application; in particular, it should not be used to pass
+	// semantic flags to a Log() implementation.
+	Payload interface{}
```

Comment:
yeah, I think that makes sense... for this PR I mainly wanted to get the interface out there for discussion, but the "semantic API" page would need to include something about this.